### PR TITLE
ROB: Do not crash when the AcroForm entry is not a dictionary

### PR DESCRIPTION
**Pull request by RavSinghChandan**
If a PDF has a /AcroForm entry that is not a dictionary, reading the form fields crashes. The code assumes the entry is a dictionary and looks for /Fields inside it. If it is a number, a string or an array instead, that check fails with a plain Python error.

get_fields already returns None when there is no /AcroForm at all, so now an entry it cannot read does the same thing and logs a warning.

This is the same problem as #4019, one level higher — that one fixed the /Fields array, this one fixes the /AcroForm entry itself.

### CODE REVIEW

**Review by stefan6419846 — CHANGES_REQUESTED** [review]
Please check the test failure (`test_workflows.py::test_get_fields_warns`).

**Comment by RavSinghChandan** [comment]
Good catch, that was mine. The old cast let a null entry through to the existing None check, but calling get_object() on it raised AttributeError. A damaged file can leave the key present with no value, which I had not tested for.

Fixed so a null entry is still treated as absent and only a wrong type is reported. I also added a test for the null case.

**Comment by RavSinghChandan** [comment]
Fixed on the 28th - a damaged file can leave the key present with no value, and calling get_object() on it raised AttributeError. Null is treated as absent again, and there is a test for it now.

File: tests/test_doc_common.py [hunk 1]
```diff
@@ -1262,3 +1262,43 @@ def test_outline__reads_a_well_formed_outlines_entry():
...
+    writer.add_blank_page(width=72, height=72)
+    writer.root_object[NameObject("/AcroForm")] = NullObject()
+    stream = BytesIO()
+    writer.write(stream)
+    stream.seek(0)
+
+    assert PdfReader(stream).get_fields() is None
+
+
+def test_get_fields__reads_a_well_formed_acro_form():
```
Comment (stefan6419846) [inline tests/test_doc_common.py]: I guess this is obsolete as well due to us already having a proper test for it?

**Review by stefan6419846 — CHANGES_REQUESTED** [review]

**Review by stefan6419846 — APPROVED** [review]
